### ISSUE TITLE
Support for .jsx extension

### ISSUE THREAD

**Issue by alexwileyy** [arminbro/generate-react-cli#18]
**Is your feature request related to a problem? Please describe.**
Currently this generates `.js` files. It would be better if it by default exported `.jsx` files as this is the filetype that's supposed to be used for React components. Could an option be added to support this?

**Describe the solution you'd like**
Either:
- JSX turned on by default
- Option in config to allow JSX export

**Comment by arminbro** [comment]
Hey @alexwileyy,

Actually the react team suggests not using the `.jsx` extension anymore. The distinction between `.js` and `.jsx` files were useful before Babel, but it’s not that useful anymore. You can read more about it [here](https://github.com/facebook/create-react-app/issues/87#issuecomment-234627904).

I'm going to close this ticket for now. I'll reopen it if you have a stronger use case for them.

**Comment by arminbro** [comment]
@alexwileyy by resolving ticket https://github.com/arminbro/generate-react-cli/issues/19, your request should also now be resolved with the new release of [v5.1.0](https://github.com/arminbro/generate-react-cli/releases/tag/v5.1.0). 

GRC now supports custom extensions via custom templates. You should now be able to declare a custom component template with its own extension (`jsx`) within your `generate-react-cli.json` file. 

**Something like this:**
```json
{
  "usesTypeScript": false,
  "usesCssModule": true,
  "cssPreprocessor": "scss",
  "testLibrary": "Testing Library",
  "component": {
    "default": {
      "customTemplates": {
        "component": "templates/components/component.jsx"
      },
      "path": "src/components",
      "withStyle": true,
      "withTest": true,
      "withStory": true,
      "withLazy": false
    }
  }
}
```

Let me know if you run into any other problems. Thanks again for the request.
